tests/manager/repositories/auth/test_auth_repository.py

```diff
@@ -11,53 +11,34 @@
 
 from ai.backend.manager.data.auth.hash import PasswordHashAlgorithm
 from ai.backend.manager.data.auth.types import GroupMembershipData, UserData
-from ai.backend.manager.errors.auth import (
-    GroupMembershipNotFoundError,
-    UserCreationError,
-)
+from ai.backend.manager.errors.auth import GroupMembershipNotFoundError, UserCreationError
 from ai.backend.manager.models.hasher.types import PasswordInfo
 from ai.backend.manager.models.user import UserRole, UserRow, UserStatus
 from ai.backend.manager.models.utils import ExtendedAsyncSAEngine
 from ai.backend.manager.repositories.auth.repository import AuthRepository
 
 
-@pytest.fixture
-def mock_db_engine():
-    """Create mocked database engine."""
-    return MagicMock(spec=ExtendedAsyncSAEngine)
-
-
-@pytest.fixture
-def mock_db_conn(mock_db_engine):
-    """Create mocked database connection with transaction support."""
-    mock_conn = AsyncMock()
-    mock_db_engine.begin.return_value.__aenter__.return_value = mock_conn
-    return mock_conn
-
-
-@pytest.fixture
-def mock_db_session(mock_db_engine):
-    """Create mocked database session."""
-    mock_session = AsyncMock()
-    mock_db_engine.begin_session.return_value.__aenter__.return_value = mock_session
-    return mock_session
-
-
-@pytest.fixture
-def mock_db_readonly_conn(mock_db_engine):
-    """Create mocked readonly database connection."""
-    mock_conn = AsyncMock()
-    mock_db_engine.begin_readonly.return_value.__aenter__.return_value = mock_conn
-    return mock_conn
-
-
 class TestAuthRepository:
     """Test cases for AuthRepository"""
 
     @pytest.fixture
-    def auth_repository(self, mock_db_engine):
+    def mock_db_engine(self) -> ExtendedAsyncSAEngine:
+        """Create mocked database engine."""
+        return MagicMock(spec=ExtendedAsyncSAEngine)
+
+    @pytest.fixture
+    def mock_db_source(self) -> AsyncMock:
+        """Create mocked AuthDBSource"""
+        return AsyncMock()
+
+    @pytest.fixture
+    def auth_repository(
+        self, mock_db_engine: ExtendedAsyncSAEngine, mock_db_source: AsyncMock
```

Comment:
It's hard to see it as mocking because it's testing by launching an actual DB container, but it seems like they're calling it a "mock DB."